x/stake/handler.go

```diff
@@ -35,6 +35,15 @@ func NewHandler(k keeper.Keeper) sdk.Handler {
 func EndBlocker(ctx sdk.Context, k keeper.Keeper) (ValidatorUpdates []abci.ValidatorUpdate) {
 	endBlockerTags := sdk.EmptyTags()
 
+	matureUnbondingValidators := k.DequeueAllMatureValidatorQueue(ctx, ctx.BlockHeader().Time)
+	for _, valAddr := range matureUnbondingValidators {
+		val, found := k.GetValidator(ctx, valAddr)
+		if !found || val.GetStatus() != sdk.Unbonding {
+			continue
+		}
+		k.UnbondingToUnbonded(ctx, val)
+	}
```

Comment:
I feel like all this logic introduced here (these ~10 lines) should be in the staking keeper, and then `UnbondingToUnbonded` would not need to be exposed